test/libsolidity/InlineAssembly.cpp

```diff
@@ -182,6 +182,18 @@ BOOST_AUTO_TEST_CASE(error_tag)
 	BOOST_CHECK(successAssemble("{ invalidJumpLabel }"));
 }
 
+BOOST_AUTO_TEST_CASE(inline_assembly_shadowed_instruction_declaration)
+{
+	// Error message: "Cannot use instruction names for identifier names."
+	BOOST_CHECK(!successAssemble("{ let gas := 1 }"));
+}
+
+BOOST_AUTO_TEST_CASE(inline_assembly_shadowed_instruction_assignment)
+{
+	// Error message: "Cannot use instruction names for identifier names."
+	BOOST_CHECK(!successAssemble("{ 2 =: gas }"));
```

Comment:
I know, but the check is only in place for `gas := 2`: https://github.com/ethereum/solidity/pull/1606/commits/f62e269115822f791a0ad2e3699815f0725a0ef7#diff-9b7bbca0dea1e2bd9ff41750a720ffe2R104